db/stats.py

```diff
@@ -73,6 +75,51 @@ def get_last_submitted_recordings():
     return last_submissions
 
 
+def calculate_stats_for_feature(path):
+    with db.engine.connect() as connection:
+        query = text("""
+            SELECT avg(x), stddev_pop(x)
+              FROM (
+            SELECT (%(path)s)::double precision AS x
+              FROM lowlevel_json
+             LIMIT :limit)
+                AS res
+        """ % {"path": path})
+        result = connection.execute(query, {'limit': NORMALIZATION_SAMPLE_SIZE})
+        return result.fetchone()
+
+
+def check_global_stats(metric_name):
+    with db.engine.connect() as connection:
+        query = text("""
+            SELECT means, stddevs
+              FROM similarity_stats
+             WHERE metric = :metric
+        """)
+        result = connection.execute(query, {"metric": metric_name})
+        return result.fetchone()
+
+
+def insert_similarity_stats(metric_name, means, stddevs):
+    with db.engine.connect() as connection:
+        query = text("""
+            INSERT INTO similarity_stats (metric, means, stddevs)
+                 VALUES (%(metric)s, %(means)s, %(stddevs)s)
```

Comment:
when inserting values you can use query parameters as well, instead of string interpolation - there are a few places where this is still used

Reply:
I've been investigating these things for the last week and for some of them I haven't been able to successfully complete the query with string interpolation, e.g. here https://pastebin.com/D0aQBj9y

I will comment some other errors that I'm getting with them as well.